### PR TITLE
Autoupdate pre-commit, update prow image tags

### PR DESCRIPTION
**Pull request by xtuchyna**
## Related Issues and Dependencies
None

## This introduces a breaking change

- [ ] Yes
- [X] No

## Description
SSIA

### CODE REVIEW

**Comment by xtuchyna** [comment]
@harshad16 do we have some issue tracking the required `autoupdate` for `pre-commit` ?

**Comment by harshad16** [comment]
> Do we have some issue tracking the required `autoupdate` for `pre-commit` ? Also for prow image tags?

No that i know about, feel free to open one @xtuchyna 
cc: @codificat

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -23,39 +23,39 @@ repos:
...
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v3.4.0
+    rev: v4.1.0
     hooks:
       - id: check-toml
       - id: check-yaml
       - id: end-of-file-fixer
       - id: trailing-whitespace
 
   - repo: https://github.com/pre-commit/mirrors-mypy
```
Comment (harshad16) [inline .pre-commit-config.yaml:38]: my suggestion for now. 
let's remove mypy out of pre-commit and can add it as a separate check like this https://github.com/thoth-station/messaging/blob/82588c945130172f0db76d81a55ea015dabea79e/.prow.yaml#L22
so our pr is not on hold just for mypy

File: Pipfile [hunk 2]
```diff
@@ -23,7 +23,12 @@ pylint = "*"
 coala = "*"
 pytest-mypy = "*"
 mypy = "*"
-pre-commit = "==2.0.1"
+pre-commit = "==2.15.0"
+types-pyyaml = "*"
+types-attrs = "*"
+stubs = "*"
+types-certifi = "*"
+types-setuptools = "*"
```
Comment (harshad16) [inline Pipfile:31]: lets remove these bit for now

Comment (harshad16) [inline Pipfile:31]: lets remove these bit for now

**Review by harshad16 — CHANGES_REQUESTED** [review]
all looks good to me
small suggestion for timebeing

**Comment by harshad16** [comment]
Already merged following commit via https://github.com/thoth-station/common/pull/1253

**Pull request closed without merging**